internal/providers/provider.go

```diff
@@ -127,6 +137,25 @@ type GetProviderSchemaResponse struct {
 	ServerCapabilities ServerCapabilities
 }
 
+// GetResourceIdentitySchemasResponse is the return type for GetProviderSchema, and
+// should only be used when handling a value for that method. The handling of
+// of schemas in any other context should always use ProviderSchema, so that
+// the in-memory representation can be more easily changed separately from the
+// RCP protocol.
```

Comment:
```suggestion
// GetResourceIdentitySchemasResponse is the return type for GetResourceIdentitySchemas, and
// should only be used when handling a value for that method. The handling of
// of schemas in any other context should always use ResourceIdentitySchemas, so that
// the in-memory representation can be more easily changed separately from the
// RPC protocol.
```